### PR TITLE
Updates dependency management

### PR DESCRIPTION
**Pull request by helrond**
- Adds dependabot configs
- Updates pre-commit configs
- Separates and updates dependencies
- Removes an unused file
- Moves sync function (model save()) out of an asynchronous function in favor of simply raising an exception.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -14,3 +14,7 @@ repos:
   rev: v4.3.0
   hooks:
   - id: isort
+- repo: https://github.com/jazzband/pip-tools
+  rev: 6.0.1
```
Comment (bonniegee) [inline .pre-commit-config.yaml]: This should likely be updated to 6.2

**Review by bonniegee — CHANGES_REQUESTED** [review]

**Review by bonniegee — APPROVED** [review]

File: requirements.in [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+djangorestframework==3.12.4
+ElectronBonder==0.7.2
+health-check==3.4.1
+iso-639==0.4.5
+jsonschema==3.2.0
+odin==1.7.0
+psycopg2-binary==2.9.1
+PyYAML==5.4.1
+rac-schemas==0.27
+requests==2.25.1
```
Comment (HaSistrunk) [inline requirements.in:14]: I think `requests` isn't top-level.

**Review by p-galligan — CHANGES_REQUESTED** [review]
I think @HaSistrunk is right on this. I built and ran tests without requests in the requirements file and it seemed to work fine.

**Comment by helrond** [comment]
@HaSistrunk and @p-galligan `requests` is a top-level dependency: https://github.com/RockefellerArchiveCenter/pisces/blob/base/fetcher/helpers.py#L1

File: requirements.in (hunk 2 above)
Comment (helrond) [reply requirements.in:14]: https://github.com/RockefellerArchiveCenter/pisces/blob/base/fetcher/helpers.py#L1

**Review by p-galligan — APPROVED** [review]

File: requirements.in (hunk 2 above)
Comment (HaSistrunk) [reply requirements.in:14]: I removed from top-level in [cartographer_backend](https://github.com/RockefellerArchiveCenter/cartographer_backend/pull/51#discussion_r664904745). Is there a different implementation context that makes it different between these repos?

**Comment by HaSistrunk** [comment]
> @HaSistrunk and @p-galligan requests is a top-level dependency: https://github.com/RockefellerArchiveCenter/pisces/blob/base/fetcher/helpers.py#L1

I removed from top-level in [cartographer_backend](https://github.com/RockefellerArchiveCenter/cartographer_backend/pull/51#discussion_r664904745). Is there a different implementation context that makes it different between these repos? Because it's imported directly here and not in that repo?

**Comment by helrond** [comment]
@HaSistrunk yes - if it's imported directly in a file then it's a top-level dependency. Otherwise, if it's used by another library but not directly, then it's a secondary dependency which we want `pip-compile` to manage for us.

The reason for this is that libraries may pin ranges of dependencies which are older than the most recent version, so if you go to try and update a secondary dependency, the application will not build. `pip-compile` helps us avoid that back and forth by minimizing the number of dependencies we need to declare and pin.

**Review by HaSistrunk — APPROVED** [review]

**Pull request merged**